modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/osgi/commands/PortalAddressOSGiCommands.java

```diff
@@ -115,24 +106,40 @@ public void repopulateCompanyCountries(long companyId) throws Exception {
 				_log.error(exception);
 			}
 		}
-	}
 
-	private void _processAdditionalCountries(Company company) throws Exception {
-		JSONArray additionalCountriesJSONArray =
-			CompanyCountriesUtil.getJSONArray(
-				"com/liferay/address/dependencies/additional-countries.json");
+		try {
+			AdditionalCountriesConfiguration additionalCountriesConfiguration =
+				_configurationProvider.getCompanyConfiguration(
+					AdditionalCountriesConfiguration.class,
+					company.getCompanyId());
 
-		for (int i = 0; i < additionalCountriesJSONArray.length(); i++) {
-			JSONObject countryJSONObject =
-				additionalCountriesJSONArray.getJSONObject(i);
+			if (additionalCountriesConfiguration.enableISOCountries()) {
+				JSONArray additionalCountriesJSONArray =
+					CompanyCountriesUtil.getJSONArray(
+						"com/liferay/address/dependencies" +
+							"/additional-countries.json");
 
-			try {
-				CompanyCountriesUtil.addCountry(
-					company, countryJSONObject, _countryLocalService,
-					_regionLocalService);
+				for (int i = 0; i < additionalCountriesJSONArray.length();
+					 i++) {
+
+					JSONObject countryJSONObject =
+						additionalCountriesJSONArray.getJSONObject(i);
+
+					countriesJSONArray.put(countryJSONObject);
```

Comment:
Is it possible to verify if the country (from the additional file) is not already available in the countriesJSONArray? Otherwise if we will update the standard one with a country that is available also in additional we will have duplicates

Reply:
Hi @smottal , 
> Is it possible to verify if the country (from the additional file) is not already available in the countriesJSONArray?
It is possible to verify the country in additional file is available in the countriesJSONArray. If the country appears in the additional, it should not appear in the default countries.